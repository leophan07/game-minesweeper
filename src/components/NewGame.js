import styled from 'styled-components';
import Button from './Button';

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  width: 200px;
  margin: 0 auto;
`;

export const PlayButton = styled(Button)`
  margin-top: 20px;
`;
