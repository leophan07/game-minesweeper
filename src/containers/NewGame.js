import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Content, PlayButton } from '../components/NewGame';

function NewGameContainer({ onPlayGame, isStarting }) {
  const [level, setLevel] = useState(9);

  return (
    <Content>
      <select onChange={(e) => setLevel(parseInt(e.target.value))}>
        <option value={9}>Beginner</option>
        <option value={16}>Advantage</option>
      </select>
      <PlayButton onClick={() => onPlayGame(level)}>
        {isStarting ? 'Please wait' : 'Play game'}
      </PlayButton>
    </Content>
  )
}

NewGameContainer.propTypes = {
  isStarting: PropTypes.bool.isRequired,
  onPlayGame: PropTypes.func.isRequired,
}

const mapStateToProps = (state) => {
  return {
    isStarting: state.isSetting,
  };
}

export default connect(mapStateToProps)(NewGameContainer);