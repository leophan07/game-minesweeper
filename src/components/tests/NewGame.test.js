import React from 'react';
import { render } from '@testing-library/react';

import { Content, PlayButton } from '../NewGame';

describe('<NewGame />', () => {

  it('Content should render an <div> tag', () => {
    const { container } = render(<Content />);
    expect(container.querySelector('div')).toBeTruthy();
  });

  it('PlayButton should render an <h1> tag', () => {
    const { container } = render(<PlayButton />);
    expect(container.querySelector('button')).toBeTruthy();
  });

  it('NewGame should match the snapshot', () => {
    const { container } = render(
      <Content>
        <PlayButton>Play Game</PlayButton>
      </Content>
    );
    expect(container).toMatchSnapshot();
  });
});
