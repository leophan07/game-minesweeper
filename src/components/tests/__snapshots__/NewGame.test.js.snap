// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewGame /> NewGame should match the snapshot 1`] = `
<div>
  <div
    class="sc-bwzfXH dMjuZz"
  >
    <button
      class="sc-bdVaJa sc-htpNat gYrVGL"
    >
      Play Game
    </button>
  </div>
</div>
`;
